### PR TITLE
Hotfix/1623 video object raises lots of warnings in logs

### PR DESCRIPTION
**Pull request by mhimon**

### CODE REVIEW

File: src/wordlift/videoobject/ajax/class-video-key-validation-service.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+			);
+		}
+
+		if ( 'youtube' === $type ) {
+			$this->validate_youtube_api_key( $api_key );
+		} else {
+			$this->validate_vimeo_api_key( $api_key );
+		}
+
+		wp_send_json_error(
```
Comment (naveen17797) [inline src/wordlift/videoobject/ajax/class-video-key-validation-service.php:64]: when this condition would be reached ?

File: src/wordlift/videoobject/ajax/class-video-key-validation-service.php [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		$response = wp_remote_get( $url );
+
+		// Get response code.
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( 200 === $response_code ) {
+			wp_send_json_success(
+				array(
+					'valid'   => true,
+					'message' => __( 'Youtube API key is valid.', 'wordlift' ),
```
Comment (naveen17797) [inline src/wordlift/videoobject/ajax/class-video-key-validation-service.php:104]: do you need to return a message on response ? are you displaying it on ui ?

File: src/wordlift/videoobject/ajax/class-video-key-validation-service.php [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	/**
+	 * Validate vimeo api key.
+	 *
+	 * @param string $api_key Vimeo api key.
+	 *
+	 * @return bool
+	 */
+	private function validate_vimeo_api_key( $api_key ) {
+		// Check if api key is empty.
+		if ( empty( $api_key ) ) {
```
Comment (naveen17797) [inline src/wordlift/videoobject/ajax/class-video-key-validation-service.php:126]: duplicated empty checks..

File: src-js/modules/video-api-key-validator.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+/**
+ * Create a Video API key validator on the element with the specified selector.
+ *
+ * @since 3.40.1
+ * @param {string} selector The element selector.
+ * @param {string} type Type.
+ */
+export const VideoAPIKeyValidator = (selector, type) => {
```
Comment (naveen17797) [inline src-js/modules/video-api-key-validator.js:16]: all the js code used for videoobject needs to be placed on one file.

File: src-js/modules/video-api-key-validator.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  selector.addEventListener("keyup", () => {
```
Comment (naveen17797) [inline src-js/modules/video-api-key-validator.js:17]: why "keyup" ? what happens if the key is pasted with a mouse event ?

File: src-js/modules/video-api-key-validator.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    type: type,
+    _wpnonce: settings["wl_video_api_nonce"],
+  })
+  .done( () => {
+    selector.classList.remove("loading");
+    selector.classList.add("valid");
+  })
+  .fail( () => {
+    selector.classList.remove("loading");
+    selector.classList.add("invalid");
```
Comment (naveen17797) [inline src-js/modules/video-api-key-validator.js:45]: does it make sense to just simplify this with replace() ?

File: src-js/screens/Admin/screens/Settings/index.js [hunk 7]
```diff
@@ -26,6 +27,21 @@ import Tabs from 'modules/tabs';
     // @since 3.32.7, country validation is disabled.
     //CountryValidator('#wl-country-code', '#wl-site-language');
 
+    const YTAPiKeySelector = document.querySelector('#wordlift_videoobject_youtube_api_key');
+    const VimeoApiSelector = document.querySelector('#wordlift_videoobject_vimeo_api_key');
+    if ( YTAPiKeySelector !== null && VimeoApiSelector !== null ) {
```
Comment (naveen17797) [inline src-js/screens/Admin/screens/Settings/index.js:32]: could this be simplified to [ ...selectors].filter(..not null).map( ..bind events) ?

File: src-js/screens/Admin/screens/Settings/styles/index.scss [hunk 8]
```diff
@@ -27,8 +32,16 @@
     // Anime.
     transition: 50ms border-color ease-in-out;
     // Background.
-    background-position: 448px 8px;
+    background-position: 98% 50%;
     background-repeat: no-repeat;
+  
+    &.loading {
+      // Icon.
+      background-image: url('loading.gif');
```
Comment (naveen17797) [inline src-js/screens/Admin/screens/Settings/styles/index.scss:40]: do you need to additional file ? is there any native wp component or loading icon from wordlift which you can reuse ?

**Review by naveen17797 — CHANGES_REQUESTED** [review]
Fix the mentioned issues.

File: src-js/screens/Admin/screens/Settings/styles/index.scss (hunk 8 above)
Comment (mhimon) [reply src-js/screens/Admin/screens/Settings/styles/index.scss:40]: didn't find anything

File: src-js/modules/video-api-key-validator.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Create a Video API key validator on the element with the specified selector.
+ *
+ * @since 3.40.1
+ * @param {string} selector The element selector.
+ * @param {string} type Type.
+ */
+export const VideoAPIKeyValidator = (selector, type) => {
+  ["keyup", "paste"].forEach( event => {
```
Comment (naveen17797) [inline src-js/modules/video-api-key-validator.js:17]: why not use "change" event here ?

**Review by naveen17797 — COMMENTED** [review]
second review

**Pull request merged**